imap_processing/codice/codice_l1a.py

```diff
@@ -324,7 +327,7 @@ def get_params(packet) -> tuple[int, int, int, int]:
     return table_id, plan_id, plan_step, view_id
 
 
-def process_codice_l1a(file_path: Path | str, data_version: str) -> xr.Dataset:
+def process_codice_l1a(file_path: Path, data_version: str) -> xr.Dataset:
```

Comment:
Is this intentionally removing `str` as an possible type?

Reply:
@sdhoyt 
Yes, because the `decom_packets(file_path)` function that is called is only specified to take a Path type. If the decom function is changed to take both Path and str then you run into the problem of having a union attribute error which would  need to be ignored for all instances (about 8). Essentially mypy would be expecting both types of the Union, Path and str in this case, to have the same attributes but they do not. So either only a Path is passed in or code would need to be added to insure that all strings passed in are then converted to Paths.